mne_bids/read.py

```diff
@@ -337,29 +337,42 @@ def read_raw_bids(bids_fname, bids_root, extra_params=None,
 
     Parameters
     ----------
-    bids_fname : str
-        Full name of the data file
+    bids_basename : str
+        The base filename of the BIDS compatible files. Typically, this can be
+        generated using :func:`mne_bids.make_bids_basename`.
     bids_root : str | pathlib.Path
         Path to root of the BIDS folder
     extra_params : None | dict
         Extra parameters to be passed to MNE read_raw_* functions.
         If a dict, for example: ``extra_params=dict(allow_maxshield=True)``.
+    kind : str | None
+        The kind of recording to read. If ``None`` and only one kind (e.g.,
+        only EEG or only MEG data) is present in the dataset, it will be
+        selected automatically.
     verbose : bool
-        The verbosity level
+        The verbosity level.
 
     Returns
     -------
     raw : instance of Raw
         The data as MNE-Python Raw object.
 
-    """
-    # Full path to data file is needed so that mne-bids knows
-    # what is the modality -- meg, eeg, ieeg to read
-    bids_fname = op.basename(bids_fname)
-    bids_basename = '_'.join(bids_fname.split('_')[:-1])
-    kind = bids_fname.split('_')[-1].split('.')[0]
-    _, ext = _parse_ext(bids_fname)
+    Raises
+    ------
+    RuntimeError
+        If multiple recording kinds are present in the dataset, but
+        ``kind=None``.
+
+    RuntimeError
+        If more than one data files exist for the specified recording.
+
+    RuntimeError
+        If no data file in a supported format can be located.
 
+    ValueError
+        If the specified ``kind`` cannot be found in the dataset.
```

Comment:
it's an interesting approach? What other repositories do this? Maybe we should fix this consistently everywhere in a separate PR if you want to do it.

Reply:
What _exactly_ do you mean? Documentation of the errors raised?

Reply:
yep!